django/contrib/auth/__init__.py

```diff
@@ -121,16 +121,22 @@ def login(request, user, backend=None):
     have to reauthenticate on every request. Note that data set during
     the anonymous session is retained when the user logs in.
     """
-    session_auth_hash = ''
+    session_auth_hash_sha256 = ''
+    session_auth_hash_sha1 = ''
     if user is None:
         user = request.user
     if hasattr(user, 'get_session_auth_hash'):
-        session_auth_hash = user.get_session_auth_hash()
```

Comment:
right now, `get_session_auth_hash` returns `sha1`, but wouldn't changing it to return `sha256` become backward incompatible ?